src/dataflow/src/render/reduce.rs

```diff
@@ -736,11 +736,22 @@ where
     G: Scope,
     G::Timestamp: Lattice,
 {
-    collection.reduce_abelian::<_, OrdValSpine<_, _, _, _>>("DistinctBy", {
-        |key, _input, output| {
-            output.push((key.clone(), 1));
+    let negated_result = collection.reduce_named("DistinctBy", {
+        |key, input, output| {
+            output.push((key.clone(), -1));
+            output.extend(
+                input
+                    .iter()
+                    .map(|(values, count)| ((*values).clone(), *count)),
+            );
         }
-    })
+    });
+    use differential_dataflow::operators::arrange::ArrangeByKey;
+    negated_result
+        .negate()
+        .concat(&collection)
+        .consolidate()
+        .arrange_by_key_named("DistinctBy")
```

Comment:
The signature expects us to return an arrangement and not a collection. Is `arrange_by_key_named` the appropriate way to convert a collection to an arrangement?

Reply:
Ah, I should have looked for this. The return of an arrangement is a "bonus" of using `reduce`, but in cases where we are not going to end on `reduce` it is probably not helpful to introduce a new `reduce`. At least, it re-introduces the cost we are hoping to avoid.

Reply:
Fixed in #7287 by modifying the return type.